非同期通信でのメッセージ送信機能実装

teru0225 opened the pull request:
WHY
非同期通信を利用することで、よりスムーズにメッセージのやりとりが可能となる。

WHAT
メッセージ送信機能を非同期通信で行えるようにした。

・massage.jsファイルの新規作成
・messages_controller,rbの編集
・create.json.jbuilderファイルの新規作成
・_main.html.hamlの編集

⬇︎フォームに文字を入力した時
https://gyazo.com/56a371e40664f3c572a0fb7457fd8cb5

⬇︎send(sumit)ボタンを押した時
https://gyazo.com/6113a239080d9a7a508d36bbcc81d488

teru0225 committed:
made new file message.js

メッセージが送信されるとイベントが起こるようにした。
Submitをクリックすることで、イベントが発生するようになっている。
このイベントが発生すると、フォームに入力された値がコントローラに送られるようになっている。
コントローラ内で、HTMLとJSONの場合で処理をわけ、jbuilderを使って返すデータを作成するようにしている。
Doneメソッドを使うことで、受け取っらデータをビューに反映するようにした。
非同期に失敗した場合には、「メッセージの送信に失敗しました」というエラー文が出るようにした。

teru0225 committed:
Edited messages_controller.rb

HTMLとJSONとで処理を分けるようにした。

teru0225 committed:
Edited _main.html.haml

非同期通信を実装する上で、この記述を追加する方が都合がいいため、この記述をした。

teru0225 committed:
made a new file create.json.jbuilder

こうすることでJavascriptファイルに返ってきたデータをjbuilderで定義したキーとバリューの形で呼び出して使うことができる。

yuki-uchida commented on app/assets/javascripts/message.js:
```diff
@@ -0,0 +1,48 @@
+$(function() {
+  function buildHTML(message){
```
buildHTMLでは、何をどうするのかがわかりづらいので、buildSendMessageHTMLとかいかがでしょうか

teru0225 replied:
かしこまりました！
buildSnedMessageHTMLに書き換えました。

yuki-uchida commented on app/assets/javascripts/message.js:
```diff
@@ -0,0 +1,48 @@
+      contentType: false
+    })
+
+    .done(function(data) {
```
dataでは、何を受け取ったのかわからないので、それがわかる名前に変えましょう

teru0225 replied:
かしこまりました！
Messagedataに書き換えました。

yuki-uchida replied:
dataという単語は無くしていいですよ。
プログラムにおいては全てがdataに当てはまってしまうためです。
ゾウという生き物に対して名前をつけるときにelephantAnimalという名前をつけるようなものです。

teru0225 replied:
> dataという単語は無くしていいですよ。
> プログラムにおいては全てがdataに当てはまってしまうためです。
> ゾウという生き物に対して名前をつけるときにelephantAnimalという名前をつけるようなものです。

その通りですね！！
Messageにします。

teru0225 committed:
Edited message.js

function buildHTML(message)
ではわかりずらいため、
function buildSendMessageHTML(message)
に変更した。

.done(function(data)
ではわかりずらいため、
.done(fucntion(Messagedata)
に変更した。

teru0225 committed:
Edited message.js

.done(function(Messagedata)
だと、分かりにくく、かつプログラムにおいては全てがdataであるため、
.done(function(message)
という記述に変えました。

yuki-uchida commented:
LGTMです。お疲れ様でした

teru0225 commented:
> LGTMです。お疲れ様でした

LGTM、かしこまりました。
レビューありがとうございました！
